README.md

```diff
@@ -72,37 +83,128 @@ You should build out the domain model in the following order:
+
+**Advanced:** Note the `app/concerns` directory. For an extra challenge, refactor your `Song`, `Artist` and `Genre` class by removing all of the common methods described above and building them into either a `Findable` module, `Creatable` module or `Destroyable` module. You should create `findable.rb`, `creatable.rb` and `destroyable.rb` files and place them in `app/concerns`.
 
 ### `LibraryParser`
 
 The `LibraryParser` should be responible for finding the MP3 files, parsing their titles, and building Song, Artist, and Genre objects from that data. 
 
-An instance of `LibraryParser` should accept a relative path from the top of the directory that points to a directory with MP3s to parse. For example, `LibraryParser.new('db/data')` would point to the `data` directory provided within `db`.
-
+<!-- An instance of `LibraryParser` should accept a relative path from the top of the directory that points to a directory with MP3s to parse. For example, `LibraryParser.new('db/data')` would point to the `data` directory provided within `db`.
+ -->
 The `library_parser_spec` defines a pretty specific vision for the library parser. It breaks it down to some small methods.
 
-`files`
+#### `.files`
+
+This method has been built for you. It sets a `@files` variables equal to the list of files contained in the `db/data` directory. Via the following line: 
+
+```ruby
+@files = Dir['db/data/*']
+```
+The [`Dir` class](http://ruby-doc.org/core-2.2.0/Dir.html) is a Ruby class that provides a variety of ways to list directories and their contents. This method returns an array of file names. 
+
+`require 'pry` at the top of the `LibraryParser` file and place a `binding.pry` into the `.files` method. Take a look at the `@files` variable to become familiar with the array you'll be operating on. 
+
+#### `.parse_filename`
+
+This method is also given to you. However, it's implementation is important to understand, so please read the explanation below. 
+
+This method operates on *a single file name*. It takes that file name in as an argument and uses a regular expression to extract from that file name the author, song and genre. Then, it returns an array that looks like this: 
+
+```ruby
+[artist_name, song_name, genre_name]
+```
+
+**Regular Expressions in Ruby:** 
```

Comment:
hmmmm is there a way we can organize the files such that they don't need regex? if the files are just in the format of `Artist Name - Song Title.mp3` then just split will be able to handle this. It's hard to kinda sorta introduce regex haha